colossus-tests/src/test/scala/colossus/core/ConnectionHandlerSpec.scala

```diff
@@ -111,34 +131,74 @@ class ConnectionHandlerSpec extends ColossusSpec {
         def connectionFailed(){}
       }
       withIOSystem{ implicit io =>
-        import RawProtocol._
         withServer(Service.become[Raw]("test", TEST_PORT){case x => x}) {
-          io ! IOCommand.BindAndConnectWorkerItem(new InetSocketAddress("localhost", TEST_PORT), new MyHandler)
-          probe.expectNoMsg(200.milliseconds)
+          io ! IOCommand.BindAndConnectWorkerItem(new InetSocketAddress("localhost", TEST_PORT), _ => new MyHandler)
+          probe.expectMsg(250.milliseconds, "UNBOUND")
+        }
+      }
+    }
+
+    "automatically unbind on disrupted connection" in {
+      val probe = TestProbe()
+      withIOSystem{ implicit io =>
+        val server = Service.become[Raw]("test", TEST_PORT){case x => x}
+        withServer(server) {
+          io ! IOCommand.BindAndConnectWorkerItem(
+            new InetSocketAddress("localhost", TEST_PORT), _ => new MyHandler(probe.ref, true, true)
+          )
+          probe.expectMsg(500.milliseconds, "BOUND")
         }
+        end(server)
+        probe.expectMsg(500.milliseconds, "UNBOUND")
+      }
+
+    }
+
+    "automatically unbind on failure to connect" in {
+      val probe = TestProbe()
+      withIOSystem{ implicit io =>
+        io ! IOCommand.BindAndConnectWorkerItem(
+          new InetSocketAddress("localhost", TEST_PORT), _ => new MyHandler(probe.ref, true, true)
```

Comment:
I"m missing something here.  What is causing the connection failure here?  This looks like the above test, sans the 'end(server)'.